controllers/stats/EditorialReportComponentHandler.inc.php

```diff
@@ -0,0 +1,239 @@
+			'SUBMISSION_DECLINED' => ['&emsp;' . __('manager.statistics.editorial.submissionsDeclined.postReview')],
+			'SUBMISSION_DECLINED_OTHER' => ['&emsp;' . __('manager.statistics.editorial.submissionsDeclined.other')],
+			'SUBMISSION_PUBLISHED' => [__('manager.statistics.editorial.submissionsPublished')],
+			'SUBMISSION_DAYS_TO_FIRST_DECIDE' => [__('manager.statistics.editorial.averageDaysToDecide'), $averageField],
+			'SUBMISSION_DAYS_TO_ACCEPT' => ['&emsp;' . __('manager.statistics.editorial.averageDaysToAccept'), $averageField],
+			'SUBMISSION_DAYS_TO_REJECT' => ['&emsp;' . __('manager.statistics.editorial.averageDaysToReject'), $averageField],
+			'SUBMISSION_ACCEPTANCE_RATE' => [__('manager.statistics.editorial.acceptanceRate'), $percentageField],
+			'SUBMISSION_REJECTION_RATE' => [__('manager.statistics.editorial.rejectionRate'), $percentageField],
+			'SUBMISSION_DECLINED_INITIAL_RATE' => ['&emsp;' . __('manager.statistics.editorial.deskRejectRate'), $percentageField],
+			'SUBMISSION_DECLINED_RATE' => ['&emsp;' . __('manager.statistics.editorial.postReviewRejectRate'), $percentageField],
+			'SUBMISSION_DECLINED_OTHER_RATE' => ['&emsp;' . __('manager.statistics.editorial.otherRejectRate'), $percentageField]
+		] as $field => list($name, $type)) {
+			$isPercentage = $type == $percentageField;
+			$isAverage = $type == $averageField || $isPercentage;
+			$editorialStatistics[] = [
+				'name' => $name,
+				'period' => round($rangedStatistics[$field], $isPercentage ? 2 : 0) . ($isPercentage ? '%' : ''),
+				'average' => round($statistics[($isAverage ? '' : 'AVG_') . $field], $isPercentage ? 2 : 0) . ($isPercentage ? '%' : ''),
+				'total' => round($statistics[$field], $isPercentage ? 2 : 0) . ($isPercentage ? '%' : '')
+			];
+		}
+		return $editorialStatistics;
+	}
+
+	/**
+	 * Given an array with statistics extracts the required user statistics to display in the component
+	 * @param $rangedStatistics array An array with a subset of the results from a call to PKPStatsService::getUserStatistics
+	 * @param $statistics array An array with the results from a call to PKPStatsService::getUserStatistics
+	 */
+	public static function extractUserStatistics($rangedStatistics, $statistics) {
+		$userStatistics = [];
+		foreach ([0 => 'manager.statistics.editorial.registeredUsers'] + Application::getRoleNames(true) as $id => $role) {
+			$userStatistics[] = [
+				'name' => __($role),
+				'period' => (int)$rangedStatistics[$id]['total'],
+				'average' => round($statistics[$id]['average']),
+				'total' => (int)$statistics[$id]['total']
+			];
+		}
+		return $userStatistics;
+	}
+
+	/**
+	 * Given an array with statistics extracts the required submission information to display in the chart component
+	 * @param $statistics array An array with the results from a call to PKPStatsService::getSubmissionStatistics
+	 */
+	public static function extractSubmissionChartData($statistics) {
```

Comment:
When you want to eliminate some duplicated code in PKP, that:
- Is completely not important
- Might be used in 2-3 different files

What do you do? Can I create a "EditorialStatisticsHelper" or something like this?

Reply:
That should go into a Service class. 

I'm trying to cut out the Helpers because what I find is that helpers proliferate, every helper has its own logic to it, and it's not easy to discover if the code you're working with has a helper or not. When you arrive to a new part of the codebase, it's very difficult to figure out where code is stored.

The Service classes are our coordinators. They contain all of the business logic related to an entity. And I'd rather have one overstuffed EditorialStatsService class than helpers because at the very least it gives us one place to look for this code. 

And putting it into a service class rather than a dedicated helper (hopefully) encourages us to abstract that code properly, rather than to just write it to solve one use-case.